modules/training.py

```diff
@@ -0,0 +1,236 @@
+import json
+import sys
+import threading
+import time
+from pathlib import Path
+
+import gradio as gr
+import torch
+import transformers
+from datasets import load_dataset
+from peft import (LoraConfig, get_peft_model, get_peft_model_state_dict,
+                  prepare_model_for_int8_training)
+
+from modules import shared, ui
+
+WANT_INTERRUPT = False
+CURRENT_STEPS = 0
+MAX_STEPS = 0
+CURRENT_GRADIENT_ACCUM = 1
+
+def get_json_dataset(path: str):
+    def get_set():
+        return ['None'] + sorted(set(map(lambda x : '.'.join(str(x.name).split('.')[:-1]), Path(path).glob('*.json'))), key=str.lower)
```

Comment:
Can't we just do this?

```python
def get_json_dataset(path: str):
    return ['None'] + sorted(set(map(lambda x : '.'.join(str(x.name).split('.')[:-1]), Path(path).glob('*.json'))), key=str.lower)
```

Reply:
Oh, you're right. Sorry that is an artifact of an older code layout that I forgot to clean up after.